src/rank_llm/rerank/pairwise/duot5.py

```diff
@@ -90,24 +101,40 @@ def create_prompt(
     ) -> Tuple[str, int]:
         query = self._replace_number(result.query.text)
 
+        reserved_for_output = (
+            64  # might need to change depending on what the actual output look like
+        )
+        query_tokens = self.get_num_tokens(
+            f"Query: {query} Document0:  Document1:  Relevant: "
+        )
+
+        few_shot_prompt = ""
+        few_shot_tokens = 0
+        few_shot_prompt = self._build_pairwise_few_shot_examples()
+        few_shot_tokens = self.get_num_tokens(few_shot_prompt)
+
+        max_token = (
+            self._context_size - reserved_for_output - query_tokens - few_shot_tokens
+        )
+
         doc1_raw = self.convert_doc_to_prompt_content(
-            result.candidates[index1].doc, max_length=self._context_size
+            result.candidates[index1].doc, max_length=max_token
         )
         doc2_raw = self.convert_doc_to_prompt_content(
-            result.candidates[index2].doc, max_length=self._context_size
+            result.candidates[index2].doc, max_length=max_token
         )
 
         doc1_tokens = self._tokenizer.encode(
-            doc1_raw, truncation=True, max_length=self._context_size
+            doc1_raw, truncation=True, max_length=max_token
         )
         doc2_tokens = self._tokenizer.encode(
-            doc2_raw, truncation=True, max_length=self._context_size
+            doc2_raw, truncation=True, max_length=max_token
         )
 
         doc1 = self._tokenizer.decode(doc1_tokens, skip_special_tokens=True)
         doc2 = self._tokenizer.decode(doc2_tokens, skip_special_tokens=True)
 
-        prompt = f"Query: {query} Document0: {doc1} Document1: {doc2} Relevant: "
+        prompt = f"{few_shot_prompt} Query: {query} Document0: {doc1} Document1: {doc2} Relevant: "
```

Comment:
Nit: please use + for string concat, this formatting will result in " Query:..." when the few_shot_prompt is an empty string